core/src/main/scala/implicits.scala

```diff
@@ -16,45 +16,43 @@
 
 package freestyle.cassandra
 
+import cats.data.Kleisli
+import cats.{~>, MonadError}
+
 import collection.JavaConverters._
 import com.datastax.driver.core._
 import com.google.common.util.concurrent._
 import freestyle._
 import freestyle.async.AsyncContext
-import freestyle.cassandra.api.LowLevelAPI
+import api.{ClusterAPI, LowLevelAPI}
+import freestyle.cassandra.api.{ClusterAPIOps, LowLevelAPIOps}
 
 object implicits {
 
-  class ListenableFutureHandler(implicit session: Session)
-      extends LowLevelAPI.Handler[ListenableFuture] {
+  class LowLevelAPIHandler[M[_]](implicit H: ListenableFuture[?] ~> M)
```

Comment:
It's just a personal preference, but I'd suggest moving these handler classes to its own package (`freestyle.cassandra.handlers`, for example).. and keep only in this `object` the implicit instances.